gcc/rust/parse/rust-parse-impl.h

```diff
@@ -1505,14 +1467,18 @@ Parser<ManagedTokenSource>::parse_macro_rules_def (AST::AttrVec outer_attrs)
 	      return std::unique_ptr<AST::MacroRulesDefinition> (
 		new AST::MacroRulesDefinition (
 		  std::move (rule_name), delim_type, std::move (macro_rules),
-		  std::move (outer_attrs), macro_locus));
+		  std::move (outer_attrs), macro_locus,
+		  AST::MacroRulesDefinition::MacroKind::MBE,
+		  AST::Visibility::create_error ()));
```

Comment:
So anytime we create an `MBE` macro, we'll have no visibility, while we should always have one when creating a `DeclMacro`. I think this means we should change the API a little bit. Instead of having one single constructor used in both situations, you can create two static functions: `AST::MacroRulesDefinition::mbe()` and `AST::MacroRulesDefinition::decl_macro()`, with the `decl_macro` taking an extra `Visibility` argument.

Both these static functions should call into the existing constructor you've modified, which can now be set to private. It's a very tiny nitpick so no worries :)